core/src/main/java/org/geysermc/geyser/item/GeyserCustomItemData.java

```diff
@@ -0,0 +1,158 @@
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * @author GeyserMC
+ * @link https://github.com/GeyserMC/Geyser
+ */
+
+package org.geysermc.geyser.item;
+
+import lombok.EqualsAndHashCode;
+import lombok.ToString;
+import org.checkerframework.checker.nullness.qual.NonNull;
+import org.geysermc.geyser.api.item.custom.CustomItemData;
+import org.geysermc.geyser.api.item.custom.CustomItemOptions;
+import org.geysermc.geyser.api.item.custom.CustomRenderOffsets;
+import org.jetbrains.annotations.NotNull;
+
+@EqualsAndHashCode
+@ToString
+public class GeyserCustomItemData implements CustomItemData {
+    private final String name;
+    private final CustomItemOptions customItemOptions;
+    private final String displayName;
+    private final String icon;
+    private final boolean allowOffhand;
+    private final int textureSize;
+    private final CustomRenderOffsets renderOffsets;
+
+    public GeyserCustomItemData(String name,
+                                CustomItemOptions customItemOptions,
+                                String displayName,
+                                String icon,
+                                boolean allowOffhand,
+                                int textureSize,
+                                CustomRenderOffsets renderOffsets) {
+        this.name = name;
+        this.customItemOptions = customItemOptions;
+        this.displayName = displayName;
+        this.icon = icon;
+        this.allowOffhand = allowOffhand;
+        this.textureSize = textureSize;
+        this.renderOffsets = renderOffsets;
+    }
+
+    public @NotNull String name() {
+        return name;
+    }
+
+    public CustomItemOptions customItemOptions() {
+        return customItemOptions;
+    }
+
+    public @NotNull String displayName() {
+        return displayName;
+    }
+
+    public @NotNull String icon() {
+        return icon;
+    }
+
+    public boolean allowOffhand() {
+        return allowOffhand;
+    }
+
+    public int textureSize() {
+        return textureSize;
+    }
+
+    public CustomRenderOffsets renderOffsets() {
+        return renderOffsets;
+    }
+
+    public static class BuilderImpl implements Builder {
```

Comment:
```suggestion
    public static class CustomItemDataBuilder implements Builder {
```

Reply:
Why? Seems like a long import/name if we ever have to use it internally.

Reply:
Consistency with the other builders.